libparsec/crates/platform_device_loader/src/web/error.rs

```diff
@@ -0,0 +1,205 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use libparsec_types::{anyhow, thiserror};
+use web_sys::wasm_bindgen::JsValue;
+
+#[derive(Debug, thiserror::Error)]
+pub enum NewStorageError {
+    #[error("No window context available, are we in a browser?")]
+    NoWindow,
+    #[error("No local storage available")]
+    NoLocalStorage,
+    #[error("Failed to access local storage ({:?})", .0)]
+    WindowError(JsValue),
+}
+
+macro_rules! impl_from_new_storage_error {
+    ($for:ty) => {
+        impl From<NewStorageError> for $for {
+            fn from(value: NewStorageError) -> Self {
+                Self::Internal(anyhow::anyhow!("{value}"))
+            }
+        }
+    };
+    ($($for:ty),*) => {
+        $(impl_from_new_storage_error!($for);)*
+    };
+}
+
+impl_from_new_storage_error!(
+    crate::LoadDeviceError,
+    crate::SaveDeviceError,
+    crate::ChangeAuthentificationError,
+    crate::ArchiveDeviceError,
+    crate::RemoveDeviceError
+);
+
+error_set::error_set! {
```

Comment:
This approach is interesting, we should definitely investigate to see how well suited it is to simplify our current cumbersome error handling system ^^

Next time, it would be better to inform us prior to introduce change that add new dependencies or challenge the standard way we do certain things :pray:

Reply:
I wanted to do an introduction on Slack but I was side tracked by more urgent issues :sweat_smile: